src/brevitas/quant_tensor/__init__.py

```diff
@@ -310,7 +310,7 @@ def cat(tensors, dim, out=None):
                 return QuantTensor(
                     value=output_value,
                     scale=output_scale,
-                    zero_point=output_zero_point,
+                    zero_point=output_zero_point.type_as(tensors[0].zero_point), # We assume all zero point have the same type, otherwise the check_zero_point fails anyway
```

Comment:
if we assume that all zero point have the same type, why are we casting it? Is "anyway" always? Comment is not really clear

Reply:
The two zero point compared have to have the same type.
This type can vary anyway (in some circumstances could be torch.int, in other torch.uin8), from what I saw during this end-to-end testing

Reply:
do you have an example for this? in principle it should always be a float

Reply:
In export mode, the zero points are cast to either int8 or uint8 (if the quantization is signed or unsigned).

After removing the line `type_as`, I wasn't able to reproduce the original error so this will be removed for the moment.